### PR TITLE
Add vhost.config for generating server configs

### PR DESCRIPTION
**Pull request by matthew-mcallister**
As part of this addition, the app vhost config file now uses the same format as the regular vhost config file.

The corresponding changes to puppet will have to wait until revisions here are done.

### CODE REVIEW

File: ocflib/vhost/config.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+import jinja2
+
+from ocflib.account.search import user_is_sorried
+from ocflib.misc.mail import email_for_user
+from ocflib.vhost.application import get_app_vhosts
+from ocflib.vhost.web import get_vhosts
+
+
+SITE_CFG = '/etc/nginx/sites-enabled/virtual'
```
Comment (chriskuehl) [inline ocflib/vhost/config.py]: i don't think ocflib show know anything about nginx... that seems like strictly application territory, not library

File: ocflib/vhost/config.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    def ssl_chain(self):
+        return '/etc/ssl/certs/lets-encrypt.crt'
+
+    @property
+    def ssl_bundle(self):
+        return str(SSL_BUNDLE / (self.fqdn + '.crt'))
+
+
+def build_web_config():
+    """Generates an apache config file for web vhosts from the config
```
Comment (chriskuehl) [inline ocflib/vhost/config.py]: i really feel this stuff doesn't belong in ocflib...

File: ocflib/vhost/vhost-app.jinja [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        {%- for alias in vhost.aliases %} "{{alias}}"{% endfor %};
+
+    location /.well-known/ {
+        proxy_pass https://www.ocf.berkeley.edu/.well-known/;
+    }
+
+    location / {
+        {% if vhost.disabled %}
+            # Proxy to the "unavailable" site, just as regular vhosts do.
+            proxy_pass https://unavailable.ocf.berkeley.edu/;
```
Comment (chriskuehl) [inline ocflib/vhost/vhost-app.jinja]: does https actually work here? the cert is invalid...

File: ocflib/vhost/vhost-app.jinja [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    {% if vhost.use_ssl %}
+        ssl on;
+        ssl_certificate {{vhost.ssl_bundle}};
+        ssl_certificate_key {{vhost.ssl_key}};
+        add_header Strict-Transport-Security "max-age=31536000";
+    {% endif %}
+}
+{% if vhost.use_ssl %}
+server {
```
Comment (chriskuehl) [inline ocflib/vhost/vhost-app.jinja]: indent?

File: ocflib/vhost/config.py (hunk 1 above)
Comment (matthew-mcallister) [inline ocflib/vhost/config.py]: This particular line was actually included by mistake.

**Comment by matthew-mcallister** [comment]
I don't remember all the details of why I originally had the idea to put all of this code in ocflib (something about deploying death on mesos), but that doesn't matter because ultimately @chriskuehl is right that most of it doesn't belong here.

For what it's worth, I don't think it would be too much of a stretch to have ocflib generate generic config files. Stuff like fastcgi and php in the apache config template, though, are way too specific to the host config to be here.

**Pull request closed without merging**